src/types.cpp

```diff
@@ -257,14 +256,11 @@ SizedType CreateCTX(size_t size, std::string name)
   return SizedType(Type::ctx, size, name);
 }
 
-SizedType CreateArray(size_t num_elem,
-                      Type elem_type,
-                      size_t elem_size,
-                      bool elem_is_signed)
+SizedType CreateArray(size_t num_elements, SizedType element_type)
```

Comment:
```suggestion
SizedType CreateArray(size_t num_elements, const SizedType &element_type)
```

We can probably take by const ref